utils/tolerance.stanza

```diff
@@ -3,36 +3,56 @@ defpackage ocdb/tolerance:
   import core
   import collections
   import math
+  import jitx/utils/logging
```

Comment:
This API is an undocumented internal library used within the JITX compiler and not public facing, it may change or be removed in the future. I think we should try and refrain from using these where possible.

Reply:
Ok, I understand, but I need a method to inform the user that a parameter that they have entered *might* be too small for the calculation that they are doing. This is important for user comprehension. Putting a fatal() in here would also *not* be a good choice. Print/println would also seem to be wrong because we want to be able at some point to log all warnings in a report.

Reply:
@m-hilgendorf Should I change it to println() for now? Or leave it as a warning() with a note that it will be improved?

@d-haldane I have used warning() to provide user feedback but still allow the compilation to continue. Sounds like we need a deeper look at the user feedback system with differentiated levels (fatal, error, warning, info). Ultimately these need to be logged somewhere so that an audit report can be created with all of the compilation information present. Just capturing stdout/stderr seems deficient for the future.

Reply:
`println` for now to get this through.

It's a good point that we need a user logging system. How do we replace the little notes people like to leave on schematics? How do we keep track of assumptions?

Reply:
The advantage to using the range [0,1] is that it is unambiguous. >1 is always incorrect in that sense. 

I agree with @d-haldane that println is appropriate here until we have a robust, general purpose logging library. We need it in any case.

Reply:
You can also println to current-error-stream()

Reply:
I think the question of [0 1] versus [0 100] can be resolved by thinking as a user when I see a function called tol%() what would I think that the argument would be? It does *not* imply right off the bat that it needs to be less than 1.0. That's the issue -> if a user tries to use the function, how will they know that the argument is in the range [0.0 1.0]? I prefer (as a user) to have an argument range that "semantically" matches the function name.

Reply:
Very good point. After a bit of thinking the alternative is still ambiguous, just in the opposite direction.

Reply:
Agreed.